print-bootp.c

```diff
@@ -993,6 +997,51 @@ rfc1048_print(netdissect_options *ndo,
 				break;
 			    }
 
+
+			case TAG_SZTP_REDIRECT:
+			    {
+				/* as per https://datatracker.ietf.org/doc/html/rfc8572#section-8.3
+				 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-...-+-+-+-+-+-+-+
+				 |	  uri-length		  |	     URI		  |
+				 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-...-+-+-+-+-+-+-+
+
+				 * uri-length: 2 octets long; specifies the length of the URI data.
+				 * URI: URI of the SZTP bootstrap server.
+				 */
+				u_int suboptnumber = 1;
+
+				while (len >= 2) {
+					suboptlen = GET_BE_U_2(bp);
+					bp += 2;
+					len -= 2;
+					ND_PRINT("\n\t	    ");
+					ND_PRINT("instance#%u: ", suboptnumber);
+					if (suboptlen == 0) {
```

Comment:
This check seems to be modelled after the `TAG_USER_CLASS` option above, which implements the requirement of RFC 3004 Section 4. However, RFC 8572 Section 8.3 does not explicitly require `uri-length` to be greater than zero. However, it does require `URI` to start with `https://`, so in any valid use case `suboptlen` would be greater than zero.

However, since this is a packet analyzer, which should be able to produce useful decoding of a packet even if the packet is malformed, it would make a bit more sense instead of skipping the rest of the option to print the empty URI and then to skip to the next URI within the same option if possible. Also it may be useful to print "invalid" or some such after every URI that does not begin with `https://`, which would also cover the empty string.